File: src - java/Esercizi/Sostituisci/Sostituisci.java
	private int _sostituisci(Nodo n, int i, int j) {
		if (n == null)
			return 0;

		//variabile accumulatore delle sostituzioni
		int sostituzioni = 0;

		// Sostituisco prima nei sottoalberi...
		if (i > n.elem)
			sostituzioni += _sostituisci(n.des, i, j);
		else
			sostituzioni += _sostituisci(n.sin, i, j);

		// ...poi nella radice
		if (n.elem == i) {
			n.elem = j;
			sostituzioni++;
		}

		return sostituzioni;
	}

	public int sostituisci(int i, int j) {
		return _sostituisci(root, i, j);
	}

File: src - c++/Esercizi/Sostituisci/Sostituisci.cpp
//Metodo privato 

unsigned int AlberoBinario::_Sostituisci(PNodo& n, TElem i, TElem j) {
  unsigned int sostituzioni = 0;  
  
  if (n) {    
    //Sostituisco prima nei sottoalberi...
    if (i > n->el)
      sostituzioni = sostituzioni + _Sostituisci(n->des, i, j);
    else
      sostituzioni = sostituzioni + _Sostituisci(n->sin, i, j);
      
    //...poi nella radice
    if (n->el == i) {
      n->el = j;
      sostituzioni++;
    }
  }
  
  return sostituzioni; 
}
  
// Metodo pubblico

unsigned int AlberoBinario::Sostituisci(TElem i, TElem j) {
  return _Sostituisci(root, i, j);
}
